osq-exts/tables/crowdstrikefalconagent/crowdstrikefalconagent.go

```diff
@@ -2,212 +2,40 @@ package crowdstrikefalconagent
-	}
-
-	stats, err := getStatsOutput(_DARWIN_FALCONCTL, _DARWIN_FALCONCTL_STATS, _DARWIN_FALCONCTL_PLIST)
-	if err != nil {
-		return prepareError(_ERR_NOT_LOADED)
-	}
-
-	parsed, err := parseRead(stats)
-	if err != nil {
-		return prepareError(_ERR_PARSE_ERROR)
-	}
-
-	return prepareResults(parsed)
+func (csfa *CrowdStrikeFalconAgent) Columns() []table.ColumnDefinition {
+	return csfa.osColumns()
 }
 
-func prepareError(reason string) ([]map[string]string, error) {
-	return []map[string]string{
-		{
-			_SENOR_OPERATIONAL: reason,
-		},
-	}, nil
+func (csfa *CrowdStrikeFalconAgent) Register() (string, []table.ColumnDefinition, table.GenerateFunc) {
+	return _PLUGIN_NAME, csfa.Columns(), csfa.Generate
 }
 
-func prepareResults(in *TopLevel) ([]map[string]string, error) {
-	return []map[string]string{
-		{
-			_AGENTID:               in.AgentInfo.AgentID,
-			_CUSTOMERID:            in.AgentInfo.CustomerID,
-			_SENOR_OPERATIONAL:     in.AgentInfo.SensorOperational,
-			_VERSION:               in.AgentInfo.Version,
-			_DC_ENABLED:            fmt.Sprintf("%v", in.DeviceControl.Enabled),
-			_DC_FSAUTH:             fmt.Sprintf("%v", in.DeviceControl.FSAuth),
-			_DC_RULECOUNT:          fmt.Sprintf("%v", in.DeviceControl.RuleCount),
-			_ES_AUTH:               fmt.Sprintf("%v", in.EndpointSecurity.Auth),
-			_ES_NOTIFY:             fmt.Sprintf("%v", in.EndpointSecurity.Notify),
-			_SA_AVG:                fmt.Sprintf("%v", in.StaticAnalysis.AVG),
-			_SA_MAX:                fmt.Sprintf("%v", in.StaticAnalysis.Max),
-			_SA_READY:              fmt.Sprintf("%v", in.StaticAnalysis.Ready),
-			_SA_REQUESTS:           fmt.Sprintf("%v", in.StaticAnalysis.Requests),
-			_SA_SUCCESSES:          fmt.Sprintf("%v", in.StaticAnalysis.Successes),
-			_BW_FILTER_CALLED:      fmt.Sprintf("%v", in.BlockWait.FilterCalled),
-			_BW_FILTER_FALSE:       fmt.Sprintf("%v", in.BlockWait.FilterFalse),
-			_BW_FILTER_SATISFIED:   fmt.Sprintf("%v", in.BlockWait.FilterSatisfied),
-			_BW_FILTER_TIMEOUTS:    fmt.Sprintf("%v", in.BlockWait.FilterTimeouts),
-			_BW_FILTER_TRUE:        fmt.Sprintf("%v", in.BlockWait.FilterTrue),
-			_BW_LATERESPONSE:       fmt.Sprintf("%v", in.BlockWait.LateResponse),
-			_BW_RECEIVER_CALLED:    fmt.Sprintf("%v", in.BlockWait.ReceiverCalled),
-			_BW_RECEIVER_SATISFIED: fmt.Sprintf("%v", in.BlockWait.ReceiverSatisfied),
-		},
-	}, nil
+func (csfa *CrowdStrikeFalconAgent) Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return csfa.osGenerate(ctx, queryContext)
 }
 
-func parseRead(in []byte) (out *TopLevel, err error) {
-	err = plist.Unmarshal(in, &out)
-	return
+func (csfa *CrowdStrikeFalconAgent) Logger() (string, logger.LogFunc) {
+	return _PLUGIN_NAME, csfa.Log
 }
 
-func checkFalconCtl(path string) (err error) {
-	_, err = os.Stat(path)
+func (csfa *CrowdStrikeFalconAgent) Log(ctx context.Context, typ logger.LogType, logText string) (err error) {
+	log.Printf("%s: %s\n", typ, logText)
```

Comment:
Remind me again where this would log to? Whatever the defined logger is (TLS endpoint or local logging subsystem)?

Reply:
This is going out to the local log for now.  I haven't yet fully explored the logging portion of it yet.  This functions more as a placeholder for me to learn and get something useful implemented.  Maybe I should mark it with a TODO comment?

Reply:
Up to you. I was mostly just curious.